kernel/src/engine/arrow_expression.rs

```diff
@@ -75,6 +75,21 @@ impl Scalar {
                 DataType::Map { .. } => unimplemented!(),
                 DataType::Struct { .. } => unimplemented!(),
             },
+            Struct(values, fields) => {
+                let mut columns = Vec::with_capacity(values.len());
+                for val in values {
+                    columns.push(val.to_array(num_rows)?);
+                }
+                Arc::new(StructArray::try_new(
+                    fields
+                        .iter()
+                        .map(TryInto::<ArrowField>::try_into)
+                        .collect::<Result<Vec<_>, _>>()?
+                        .into(),
```

Comment:
qq: What decides imperative (loop) vs. declarative (iterator) here? Also, [Itertools::try_collect](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.try_collect)?
```suggestion
let columns = values.iter().map(|val| val.to_array(num_rows)).try_collect()?;
let fields = fields.iter().map(ArrowField::try_from).try_collect()?;
Arc::new(StructArray::try_new(
    fields.into(),
    columns,
```
(depending on the compiler's mood, `fields` may need a `: Vec<_> = ` type annotation.. but I _think_ it's unnecessary here)

Reply:
yeah - I think we stumbled across this before. The reason for the loop vs. iterator was to pre-allocate the vector. IIRC, we did land on doing iterators since this optimization is usually applied internally in `collect` as well and the nicer syntax.

In any case - will update.